src/allotropy/parsers/moldev_softmax_pro/softmax_pro_parser.py

```diff
@@ -71,10 +80,14 @@ def _get_model(self, file_name: str, data: Data) -> Model:
                 ),
                 plate_reader_document=[
                     self._get_plate_reader_document_item(plate_block, position)
-                    for plate_block in data.get_plate_block()
+                    for plate_block in data.block_list.plate_blocks.values()
                     for position in plate_block.iter_wells()
                 ],
-                calculated_data_aggregate_document=None,
+                calculated_data_aggregate_document=CalculatedDataAggregateDocument(
+                    calculated_data_document=list(
```

Comment:
Similar to comments in #217, let's have the helper return a list rather than an Iterator unless you see a near-term future need for an iterator.

P.S. Although in this case, your get-iterator method has more than one `yield`, so maybe it's fine as is?

Reply:
There are two types of calculated data documents, the first one is from the reduced data in the plate block and the second one is from the group block. The function iter_calculated_data_documents iterates over all of them, that is why is an iterator. However, It can also be implemented as a list